src/tests/ftest/util/test_utils_pool.py

```diff
@@ -907,40 +907,37 @@ def check_pool_daos_space(self, s_total=None, s_free=None):
             for index, item in enumerate(val)]
         return self._check_info(checks)
 
-    def check_free_space(self, expected_scm=None, expected_nvme=None, timeout=30):
+    def check_free_space(self, expected_scm=None, expected_nvme=None, timeout=30, interval=1):
         """Check pool free space with expected value.
 
         Args:
             expected_scm (int, optional): pool expected SCM free space.
             expected_nvme (int, optional): pool expected NVME free space.
-            timeout(int, optional): time to fail test if it could not match
+            timeout (int, optional): time to fail test if it could not match
                 expected values.
+            interval (int, optional): time to sleep between retries
 
         Note:
             Arguments may also be provided as a string with a number preceded
             by '<', '<=', '>', or '>=' for other comparisons besides the
             default '=='.
 
         Raises:
-            DaosTestError: if scm or nvme free space doesn't match expected
-            values within timeout.
+            ValueError: if no space arguments are given
 
         Returns:
-            bool: True if expected value is specified and all the
-                specified values match; False if no space parameters specified.
+            bool: whether the space matched expected values
 
         """
         if not expected_scm and not expected_nvme:
-            self.log.error("at least one space parameter must be specified")
-            return False
+            raise ValueError("at least one space parameter must be specified")
 
-        done = False
         scm_fs = 0
         nvme_fs = 0
         start = time()
         scm_index, nvme_index = 0, 1
-        while time() - start < timeout and not done:
-            sleep(1)
+        while time() - start < timeout:
```

Comment:
For this type of loop - that we run in multiple places - I wonder if we should start to use https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/general_utils.py#L1136 more often.  To replicate this exact behavior it may need a silent option though.

Reply:
I'm generally not a fan of the logging in that function. It makes it messy to read the log, especially for cases where the interval is low